bundles/binding/org.openhab.binding.fatekplc/src/main/java/org/openhab/binding/fatekplc/internal/FatekPLCSlave.java

```diff
@@ -0,0 +1,162 @@
+import com.github.s4u.jfatek.FatekException;
+import com.github.s4u.jfatek.FatekPLC;
+import com.github.s4u.jfatek.FatekReadMixDataCmd;
+import com.github.s4u.jfatek.io.FatekIOException;
+import com.github.s4u.jfatek.registers.Reg;
+import com.github.s4u.jfatek.registers.RegValue;
+
+/**
+ * Implementation of one fatek slave.
+ *
+ * @author Slawomir Jaranowski
+ * @since 1.9.0
+ */
+public class FatekPLCSlave {
+
+	/**
+	 * Main logger.
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(FatekPLCSlave.class);
+
+	// OSGI Service
+	private ItemRegistry itemRegistry;
+
+	private EventPublisher eventPublisher;
+
+	private String name;
+
+	private FatekPLC fatekPLC = null;
+
+	public FatekPLCSlave(String name, EventPublisher eventPublisher, ItemRegistry itemRegistry) {
+		this.name = name;
+		this.eventPublisher = eventPublisher;
+		this.itemRegistry = itemRegistry;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void disconnect() throws FatekIOException {
+
+		if (fatekPLC != null) {
+			fatekPLC.close();
+		}
+	}
+
+	/**
+	 * Configure slave property
+	 *
+	 * @param stri
+	 * @param object
+	 * @throws FatekIOException
+	 */
+	public void configure(String name, Object value) throws ConfigurationException {
+
+		logger.debug("Configure item: {} to {}", name, value);
+
+		try {
+			if ("connectionUri".equals(name)) {
+				try {
+					disconnect();
+					fatekPLC = new FatekPLC((String) value);
+					logger.info("New Fatek PLC connectionUri={}", value);
+				} catch (Exception e) {
+					fatekPLC = null;
+					throw new ConfigurationException(name, e.getMessage());
+				}
+			} else {
+				throw new ConfigurationException(name, "Unknow property");
+			}
+		} catch (ConfigurationException e) {
+			logger.error("", e);
+			throw e;
+		}
+		logger.debug("Configure item: {} end", name);
+	}
+
+	/**
+	 * Read current data from Fatek PLC and publish new state on OpenHab bus.
+	 *
+	 * @param eventPublisher
+	 * @param items
+	 * @throws FatekIOException
+	 * @throws FatekException
+	 */
+	public void updateItems(List<FatekPLCItem> items) throws FatekIOException, FatekException {
+
+		// collect unique register to read from PLC
+		Set<Reg> regsToUpdate = new HashSet<>();
+		for (FatekPLCItem item : items) {
+			regsToUpdate.addAll(item.getRegs());
+		}
+
+		Map<Reg, RegValue> response;
+		synchronized (fatekPLC) {
+			// read register
+			response = new FatekReadMixDataCmd(fatekPLC, regsToUpdate).send();
+		}
+
+		// distribute read data
+		for (FatekPLCItem fatekItem : items) {
+
+			State newState = fatekItem.getState(response);
+			State currentState;
+			Item item;
+			try {
+				item = itemRegistry.getItem(fatekItem.getItemName());
+				currentState = item.getState();
+			} catch (ItemNotFoundException e) {
+				logger.warn("", e);
+				currentState = null;
+				item = null;
+			}
+
+			logger.debug("new state: {}, current state: {}, item: {}", newState, currentState, item);
+			if (!newState.equals(currentState)) {
```

Comment:
If it would be common to post an update for items ten times per second unless "no change" updates were filtered out, then yes, you would probably want to avoid that!  But don't let me discourage you from pursuing real data on which approach is best! 😄